src/db/mysql/models.rs

```diff
@@ -484,10 +484,11 @@ impl MysqlDb {
         }
 
         query = match sort {
-            Sorting::Index => query.order(bso::sortindex.desc()),
-            Sorting::Newest => query.order(bso::modified.desc()),
-            Sorting::Oldest => query.order(bso::modified.asc()),
-            _ => query,
+            Sorting::Index => query.order(bso::id.desc()).order(bso::sortindex.desc()),
```

Comment:
A tuple also works FYI:

```suggestion
            Sorting::Index => query.order((bso::sortindex.desc(), bso::id.desc())),
```